sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/UnivocityParser.scala

```diff
@@ -183,11 +183,19 @@ class UnivocityParser(
     }
   }
 
+  private lazy val doParse = if (schema.nonEmpty) {
```

Comment:
Do you really need lazy here. In most cases, time interval between calling of the constructor and the parse() method is pretty short. I don't think we win something here.

Reply:
recheck: it seems `parse` is called in executor sides only, so we can add `lazy` here to avoid unnecessary instantiation?